### ISSUE TITLE
ERROR: Unable to extract 'data.connectionData.externApValue

### ISSUE THREAD

**Issue by todrees1967** [bb-Ricardo/fritzinfluxdb#47]
Hi, when running the script, I am facing 2 error messages:

ERROR: Unable to extract 'data.connectionData.externApValue
ERROR: Unable to extract 'data.connectionData.version

Nevertheless the InfluxDB is filled with data.

This is the full log:

```
2022-08-20 14:54:23,149 - INFO: Starting fritzinfluxdb v1.0.0 (2022-06-11)
2022-08-20 14:54:23,150 - INFO: Done reading config files
2022-08-20 14:54:23,196 - INFO: Successfully parsed config
2022-08-20 14:54:23,204 - INFO: Connection to InfluxDB 1.8.6 established and database present
2022-08-20 14:54:24,240 - INFO: Successfully established FritzBox TR-069 session
2022-08-20 14:54:24,383 - INFO: Successfully established FritzBox Lua session
2022-08-20 14:54:24,383 - INFO: Successfully connected to FritzBox '192.168.111.1' (fritz.box) Model: FRITZ!Box 6591 Cable (kdg) - FW: 7.29
2022-08-20 14:54:24,385 - INFO: Starting main loop
2022-08-20 14:54:24,386 - INFO: Requested invalid service: WANCommonIFC
2022-08-20 14:54:24,386 - INFO: Querying service 'WANCommonIFC' will be disabled
2022-08-20 14:54:24,386 - INFO: Requested invalid service: WANIPConn
2022-08-20 14:54:24,386 - INFO: Querying service 'WANIPConn' will be disabled
2022-08-20 14:54:24,603 - INFO: Requested invalid action 'GetInfo' for service: WANDSLInterfaceConfig
2022-08-20 14:54:24,603 - INFO: Querying action 'GetInfo' will be disabled
2022-08-20 14:54:24,615 - INFO: Requested invalid action 'GetStatisticsTotal' for service: WANDSLInterfaceConfig
2022-08-20 14:54:24,615 - INFO: Querying action 'GetStatisticsTotal' will be disabled
2022-08-20 14:54:24,627 - INFO: Requested invalid action 'X_AVM-DE_GetDSLInfo' for service: WANDSLInterfaceConfig
2022-08-20 14:54:24,628 - INFO: Querying action 'X_AVM-DE_GetDSLInfo' will be disabled
2022-08-20 14:54:24,628 - INFO: All actions for service 'WANDSLInterfaceConfig' are unavailable. Disabling service.
2022-08-20 14:54:24,681 - INFO: Requested invalid action 'GetInfo' for service: WANPPPConnection:1
2022-08-20 14:54:24,682 - INFO: Querying action 'GetInfo' will be disabled
2022-08-20 14:54:24,682 - INFO: All actions for service 'WANPPPConnection:1' are unavailable. Disabling service.
2022-08-20 14:54:31,595 - INFO: Unable to request FritzBox Lua service 'DSL Info'
2022-08-20 14:54:31,595 - INFO: FritzBox Lua service 'DSL Info' will be disabled.
2022-08-20 14:54:31,820 - ERROR: Unable to extract 'data.connectionData.externApValue' form '{'pid': 'docOv', 'hide': {'shareUsb': True, 'liveTv': True, 'tvsd': True, 'dvbSet': True, 'mobile': True, 'tvhd': True, 'ssoSet': True, 'dvbSig': True, 'dvbradio': True}, 'time': [], 'data': {'connectionData': {'frequencyScantext': 'Verbindungsaufbau läuft... Dieser Vorgang kann je nach Anschluss bis zu 5 Minuten dauern.', 'debugidx': 0, 'fullLine': False, 'dsCount': 32, 'usCount': 4, 'isDebug': False, 'externAPHeader': 'Kopfstelle (CMTS)', 'dsCountSecond': 1, 'line': [{'state': 'ready', 'timePrefix': 'Verbindungsdauer:', 'trainState': 'aktiv', 'mode': 'DOCSIS 3.0 und DOCSIS 3.1', 'trainStatePrefix': 'Kabel-Internet:', 'time': '288 Stunden 44 Minuten 55 Sekunden'}], 'usCountSecond': 1, 'showCableImg': True, 'usFreqs': {'max': 85, 'values': {'docsis3_1': [[29, 64]], 'docsis3_0': [51, 30, 37, 44]}, 'min': 5}, 'dsRate': '1150 Mbit/s', 'usRate': '56,7 Mbit/s', 'dsFreqs': {'max': 1218, 'values': {'docsis3_1': [[151, 324]], 'docsis3_0': [642, 114, 130, 138, 146, 602, 618, 626, 650, 658, 666, 674, 682, 690, 698, 706, 714, 722, 730, 738, 746, 754, 762, 770, 778, 786, 794, 802, 810, 818, 826, 834]}, 'min': 108}}}, 'sid': 'bfb40e062f09c063'}', got '<class 'NoneType'>'
2022-08-20 14:54:31,821 - ERROR: Unable to extract 'data.connectionData.version' form '{'pid': 'docOv', 'hide': {'shareUsb': True, 'liveTv': True, 'tvsd': True, 'dvbSet': True, 'mobile': True, 'tvhd': True, 'ssoSet': True, 'dvbSig': True, 'dvbradio': True}, 'time': [], 'data': {'connectionData': {'frequencyScantext': 'Verbindungsaufbau läuft... Dieser Vorgang kann je nach Anschluss bis zu 5 Minuten dauern.', 'debugidx': 0, 'fullLine': False, 'dsCount': 32, 'usCount': 4, 'isDebug': False, 'externAPHeader': 'Kopfstelle (CMTS)', 'dsCountSecond': 1, 'line': [{'state': 'ready', 'timePrefix': 'Verbindungsdauer:', 'trainState': 'aktiv', 'mode': 'DOCSIS 3.0 und DOCSIS 3.1', 'trainStatePrefix': 'Kabel-Internet:', 'time': '288 Stunden 44 Minuten 55 Sekunden'}], 'usCountSecond': 1, 'showCableImg': True, 'usFreqs': {'max': 85, 'values': {'docsis3_1': [[29, 64]], 'docsis3_0': [51, 30, 37, 44]}, 'min': 5}, 'dsRate': '1150 Mbit/s', 'usRate': '56,7 Mbit/s', 'dsFreqs': {'max': 1218, 'values': {'docsis3_1': [[151, 324]], 'docsis3_0': [642, 114, 130, 138, 146, 602, 618, 626, 650, 658, 666, 674, 682, 690, 698, 706, 714, 722, 730, 738, 746, 754, 762, 770, 778, 786, 794, 802, 810, 818, 826, 834]}, 'min': 108}}}, 'sid': 'bfb40e062f09c063'}', got '<class 'NoneType'>'
```

What is causing this error?
Regards,
Torsten

**Comment by bb-Ricardo** [comment]
Hi Torsten,

This should not happen. 
I guess the values for the cable connection are a bit different for your fritzbox. Will have a look later today, hopefully.

**Comment by todrees1967** [comment]
> Hi Torsten,
> 
> This should not happen. I guess the values for the cable connection are a bit different for your fritzbox. Will have a look later today, hopefully.
> 
> As far as I can see these values are missing from the. FritzBox output. Is your cable connection established? From the info it seems to try to connect.

Hi Ricardo, yes, of course my cable connection is established. But you are right, the german messages inside the error indicate that the box tries to establish a connection. That message appears every 10 minutes ...

**Comment by bb-Ricardo** [comment]
Hi, I just pushed a new release. Can you try to update and see if the error is fixed?

Thank you.

**Comment by todrees1967** [comment]
Hi, I have updated, but I still get the same error:

2022-08-23 09:14:08,575 - INFO: Starting fritzinfluxdb v1.0.1 (2022-08-22)
2022-08-23 09:14:08,577 - INFO: Done reading config files
2022-08-23 09:14:08,622 - INFO: Successfully parsed config
2022-08-23 09:14:08,630 - INFO: Connection to InfluxDB 1.8.6 established and database present
2022-08-23 09:14:10,245 - INFO: Successfully established FritzBox TR-069 session
2022-08-23 09:14:10,387 - INFO: Successfully established FritzBox Lua session
2022-08-23 09:14:10,387 - INFO: Successfully connected to FritzBox '192.168.111.1' (fritz.box) Model: FRITZ!Box 6591 Cable (kdg) - FW: 7.29
2022-08-23 09:14:10,389 - INFO: Starting main loop
2022-08-23 09:14:10,753 - INFO: Requested invalid action 'GetInfo' for service: WANDSLInterfaceConfig
2022-08-23 09:14:10,753 - INFO: Querying action 'GetInfo' will be disabled
2022-08-23 09:14:10,765 - INFO: Requested invalid action 'GetStatisticsTotal' for service: WANDSLInterfaceConfig
2022-08-23 09:14:10,765 - INFO: Querying action 'GetStatisticsTotal' will be disabled
2022-08-23 09:14:10,774 - INFO: Requested invalid action 'X_AVM-DE_GetDSLInfo' for service: WANDSLInterfaceConfig
2022-08-23 09:14:10,774 - INFO: Querying action 'X_AVM-DE_GetDSLInfo' will be disabled
2022-08-23 09:14:10,774 - INFO: All actions for service 'WANDSLInterfaceConfig' are unavailable. Disabling service.
2022-08-23 09:14:10,828 - INFO: Requested invalid action 'GetInfo' for service: WANPPPConnection:1
2022-08-23 09:14:10,828 - INFO: Querying action 'GetInfo' will be disabled
2022-08-23 09:14:10,828 - INFO: All actions for service 'WANPPPConnection:1' are unavailable. Disabling service.
2022-08-23 09:14:18,138 - INFO: Unable to request FritzBox Lua service 'DSL Info'
2022-08-23 09:14:18,138 - INFO: FritzBox Lua service 'DSL Info' will be disabled.
2022-08-23 09:14:18,350 - ERROR: Unable to extract 'data.connectionData.externApValue' form '{'pid': 'docOv', 'hide': {'shareUsb': True, 'liveTv': True, 'tvsd': True, 'dvbSet': True, 'mobile': True, 'tvhd': True, 'ssoSet': True, 'dvbSig': True, 'dvbradio': True}, 'time': [], 'data': {'connectionData': {'frequencyScantext': 'Verbindungsaufbau läuft... Dieser Vorgang kann je nach Anschluss bis zu 5 Minuten dauern.', 'debugidx': 0, 'fullLine': False, 'dsCount': 32, 'usCount': 4, 'isDebug': False, 'externAPHeader': 'Kopfstelle (CMTS)', 'dsCountSecond': 1, 'line': [{'state': 'ready', 'timePrefix': 'Verbindungsdauer:', 'trainState': 'aktiv', 'mode': 'DOCSIS 3.0 und DOCSIS 3.1', 'trainStatePrefix': 'Kabel-Internet:', 'time': '5 Stunden 47 Minuten 35 Sekunden'}], 'usCountSecond': 1, 'showCableImg': True, 'usFreqs': {'max': 85, 'values': {'docsis3_1': [[29, 64]], 'docsis3_0': [44, 30, 37, 51]}, 'min': 5}, 'dsRate': '1150 Mbit/s', 'usRate': '56,7 Mbit/s', 'dsFreqs': {'max': 1218, 'values': {'docsis3_1': [[151, 324]], 'docsis3_0': [666, 114, 130, 138, 146, 602, 618, 626, 642, 650, 658, 674, 682, 690, 698, 706, 714, 722, 730, 738, 746, 754, 762, 770, 778, 786, 794, 802, 810, 818, 826, 834]}, 'min': 108}}}, 'sid': '1385989e2ce01240'}', got '<class 'NoneType'>'
2022-08-23 09:14:18,351 - ERROR: Unable to extract 'data.connectionData.version' form '{'pid': 'docOv', 'hide': {'shareUsb': True, 'liveTv': True, 'tvsd': True, 'dvbSet': True, 'mobile': True, 'tvhd': True, 'ssoSet': True, 'dvbSig': True, 'dvbradio': True}, 'time': [], 'data': {'connectionData': {'frequencyScantext': 'Verbindungsaufbau läuft... Dieser Vorgang kann je nach Anschluss bis zu 5 Minuten dauern.', 'debugidx': 0, 'fullLine': False, 'dsCount': 32, 'usCount': 4, 'isDebug': False, 'externAPHeader': 'Kopfstelle (CMTS)', 'dsCountSecond': 1, 'line': [{'state': 'ready', 'timePrefix': 'Verbindungsdauer:', 'trainState': 'aktiv', 'mode': 'DOCSIS 3.0 und DOCSIS 3.1', 'trainStatePrefix': 'Kabel-Internet:', 'time': '5 Stunden 47 Minuten 35 Sekunden'}], 'usCountSecond': 1, 'showCableImg': True, 'usFreqs': {'max': 85, 'values': {'docsis3_1': [[29, 64]], 'docsis3_0': [44, 30, 37, 51]}, 'min': 5}, 'dsRate': '1150 Mbit/s', 'usRate': '56,7 Mbit/s', 'dsFreqs': {'max': 1218, 'values': {'docsis3_1': [[151, 324]], 'docsis3_0': [666, 114, 130, 138, 146, 602, 618, 626, 642, 650, 658, 674, 682, 690, 698, 706, 714, 722, 730, 738, 746, 754, 762, 770, 778, 786, 794, 802, 810, 818, 826, 834]}, 'min': 108}}}, 'sid': '1385989e2ce01240'}', got '<class 'NoneType'>'

Regards,
Torsten

**Comment by bb-Ricardo** [comment]
Thank you for the update, will try to add a fix later today.

**Comment by todrees1967** [comment]
If I can help anything in regard of testing or supplying more information,
just let me know.

Viele Grüße

Torsten Drees

gesendet mit meinem Commodore C64

Ricardo ***@***.***> schrieb am Mi., 24. Aug. 2022, 12:03:

**Comment by bb-Ricardo** [comment]
Hi Torsten,

I just pushed a change to the `next-release` branch. A copy-paste error probably didn't fix your issue in 1.0.1.

I somehow doubt that you are sending emails from a C64 but if you do, consider me impressed.

**Comment by todrees1967** [comment]
Hi Ricardo,

i have checked that branch and I do no longer get an error. This is the
output from the console:

***@***.***:/opt/fritzinfluxdb$ ./fritzinfluxdb.py
2022-08-24 18:10:08,541 - INFO: Starting fritzinfluxdb v1.0.1 (2022-08-22)
2022-08-24 18:10:08,543 - INFO: Done reading config files
2022-08-24 18:10:08,588 - INFO: Successfully parsed config
2022-08-24 18:10:08,596 - INFO: Connection to InfluxDB 1.8.6 established
and database present
2022-08-24 18:10:09,272 - INFO: Successfully established FritzBox TR-069
session
2022-08-24 18:10:09,425 - INFO: Successfully established FritzBox Lua
session
2022-08-24 18:10:09,425 - INFO: Successfully connected to FritzBox
'192.168.111.1' (fritz.box) Model: FRITZ!Box 6591 Cable (kdg) - FW: 7.29
2022-08-24 18:10:09,428 - INFO: Starting main loop
2022-08-24 18:10:09,794 - INFO: Requested invalid action 'GetInfo' for
service: WANDSLInterfaceConfig
2022-08-24 18:10:09,795 - INFO: Querying action 'GetInfo' will be disabled
2022-08-24 18:10:09,805 - INFO: Requested invalid action
'GetStatisticsTotal' for service: WANDSLInterfaceConfig
2022-08-24 18:10:09,806 - INFO: Querying action 'GetStatisticsTotal' will
be disabled
2022-08-24 18:10:09,817 - INFO: Requested invalid action
'X_AVM-DE_GetDSLInfo' for service: WANDSLInterfaceConfig
2022-08-24 18:10:09,817 - INFO: Querying action 'X_AVM-DE_GetDSLInfo' will
be disabled
2022-08-24 18:10:09,818 - INFO: All actions for service
'WANDSLInterfaceConfig' are unavailable. Disabling service.
2022-08-24 18:10:09,871 - INFO: Requested invalid action 'GetInfo' for
service: WANPPPConnection:1
2022-08-24 18:10:09,871 - INFO: Querying action 'GetInfo' will be disabled
2022-08-24 18:10:09,872 - INFO: All actions for service
'WANPPPConnection:1' are unavailable. Disabling service.
2022-08-24 18:10:16,800 - INFO: Unable to request FritzBox Lua service 'DSL
Info'
2022-08-24 18:10:16,801 - INFO: FritzBox Lua service 'DSL Info' will be
disabled.


So it seems the problem has been solved. I did not check, if data still
arrives in the Influx DB, since I do not see an error here, i assume this
is OK. I'll check that later, when I am at my desk (now using my tablet).

And of course, you are right, I am not using a C64, this is only a joke,
inspired by all the ones having "Sent from my iPhone" in their footer ....

Viele Grüße

Torsten

Gesendet mit meinem Commodore C64

Ricardo ***@***.***> schrieb am Mi., 24. Aug. 2022, 17:18:
